src/couch_query_servers.erl

```diff
@@ -296,7 +296,8 @@ filter_docs(Req, Db, DDoc, FName, Docs) ->
     #httpd{} = HttpReq ->
         couch_httpd_external:json_req_obj(HttpReq, Db)
     end,
-    JsonDocs = [couch_doc:to_json_obj(Doc, [revs]) || Doc <- Docs],
+    Limit = config:get_integer("query_server_config", "filter_revs_limit", 20),
```

Comment:
@kxepal I looked at all the places that call `couch_query_server:json_doc/1` and I think you're right, that would be a good safe option to making the `revs_limit` logic work here for show, filter, and update functions. Any objections @davisp @rnewson?